airbyte-integrations/connectors/source-google-sheets/source_google_sheets/source.py

```diff
@@ -165,46 +169,61 @@ def _read(
         logger.info(f"Row counts: {sheet_row_counts}")
         for sheet in sheet_to_column_index_to_name.keys():
             logger.info(f"Syncing sheet {sheet}")
-            stream = stream_name_to_stream.get(sheet).stream
+            stream = stream_instances.get(sheet)
             yield as_airbyte_message(stream, AirbyteStreamStatus.STARTED)
+            checkpoint_reader = FullRefreshCheckpointReader([])
+            _ = checkpoint_reader.next()
             # We revalidate the sheet here to avoid errors in case the sheet was changed after the sync started
             is_valid, reason = Helpers.check_sheet_is_valid(client, spreadsheet_id, sheet)
-            if is_valid:
-                column_index_to_name = sheet_to_column_index_to_name[sheet]
-                row_cursor = 2  # we start syncing past the header row
-                # For the loop, it is necessary that the initial row exists when we send a request to the API,
-                # if the last row of the interval goes outside the sheet - this is normal, we will return
-                # only the real data of the sheet and in the next iteration we will loop out.
-                while row_cursor <= sheet_row_counts[sheet]:
-                    row_batch = SpreadsheetValues.parse_obj(
-                        client.get_values(
-                            sheet=sheet,
-                            row_cursor=row_cursor,
-                            spreadsheetId=spreadsheet_id,
-                            majorDimension="ROWS",
-                        )
+            if not is_valid:
+                logger.info(f"Skipping syncing sheet {sheet}: {reason}")
+                yield self._checkpoint_state(checkpoint_reader.get_checkpoint(), state_manager, sheet, None)
+                yield as_airbyte_message(stream, AirbyteStreamStatus.COMPLETE)
```

Comment:
should we yield complete message here?

Reply:
`incomplete` is a more suitable option here, thanks